src/Features/Core/Portable/ConflictMarkerResolution/AbstractConflictMarkerCodeFixProvider.cs

```diff
@@ -109,27 +123,66 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
             endLine = default;
 
             var lines = text.Lines;
+            bool foundBarLine;
+            bool success;
             switch (text[position])
             {
                 case '<':
                     startLine = lines.GetLineFromPosition(position);
-                    return TryFindLineForwards(startLine, '=', out middleLine) &&
-                           TryFindLineForwards(middleLine, '>', out endLine);
+                    foundBarLine = TryFindLineForwards(startLine, '|', out firstMiddleLine);
+
+                    success = TryFindLineForwards(foundBarLine ? firstMiddleLine : startLine, '=', out secondMiddleLine) &&
+                              TryFindLineForwards(secondMiddleLine, '>', out endLine);
+
+                    if (success && !foundBarLine)
+                    {
+                        firstMiddleLine = secondMiddleLine;
+                    }
+
+                    return success;
+                case '|':
+                    firstMiddleLine = lines.GetLineFromPosition(position);
+                    return TryFindLineBackwards(firstMiddleLine, '<', out startLine) &&
+                           TryFindLineForwards(firstMiddleLine, '=', out secondMiddleLine) &&
+                           TryFindLineForwards(secondMiddleLine, '>', out endLine);
                 case '=':
-                    middleLine = lines.GetLineFromPosition(position);
-                    return TryFindLineBackwards(middleLine, '<', out startLine) &&
-                           TryFindLineForwards(middleLine, '>', out endLine);
+                    secondMiddleLine = lines.GetLineFromPosition(position);
+                    foundBarLine = TryFindLineBackwards(secondMiddleLine, '|', out firstMiddleLine);
+
+                    success = TryFindLineBackwards(foundBarLine ? firstMiddleLine : secondMiddleLine, '<', out startLine) &&
+                              TryFindLineForwards(secondMiddleLine, '>', out endLine);
+
+                    if (success && !foundBarLine)
+                    {
+                        firstMiddleLine = secondMiddleLine;
+                    }
+
+                    return success;
                 case '>':
                     endLine = lines.GetLineFromPosition(position);
-                    return TryFindLineBackwards(endLine, '=', out middleLine) &&
-                           TryFindLineBackwards(middleLine, '<', out startLine);
+                    if (!TryFindLineBackwards(endLine, '=', out secondMiddleLine))
+                    {
+                        return false;
+                    }
+
+                    foundBarLine = TryFindLineBackwards(secondMiddleLine, '|', out firstMiddleLine);
+
+                    success = TryFindLineBackwards(secondMiddleLine, '<', out startLine);
+
+                    if (success && !foundBarLine)
+                    {
+                        firstMiddleLine = secondMiddleLine;
+                    }
+
+                    return success;
```

Comment:
nit: all these blocks end teh same way.  so instead, can you break out and then handle the `firstMiddleLien = secondMiddleLine` stuff outside the switch?  thanks!